Reduce use of LegacyNullableObjectIdentifier

cdumez opened the pull request:
#### 584c5f30a3a23301b7464d7c77a03853460b4b92
<pre>
Reduce use of LegacyNullableObjectIdentifier
<a href="https://bugs.webkit.org/show_bug.cgi?id=278298">https://bugs.webkit.org/show_bug.cgi?id=278298</a>

Reviewed by Per Arne Vollan.

Reduce use of LegacyNullableObjectIdentifier, and use ObjectIdentifier instead.

* Source/WTF/wtf/HashMap.h:
(WTF::Y&gt;::takeOptional):
* Source/WebCore/html/track/InbandGenericTextTrack.cpp:
(WebCore::GenericTextTrackCueMap::remove):
* Source/WebCore/page/cocoa/ImageOverlayDataDetectionResultIdentifier.h:
* Source/WebCore/page/mac/ImageOverlayControllerMac.mm:
(WebCore::ImageOverlayController::handleDataDetectorAction):
* Source/WebCore/platform/graphics/ImageDecoderIdentifier.h:
* Source/WebCore/platform/graphics/InbandGenericCue.cpp:
(WebCore::InbandGenericCue::toJSONString const):
* Source/WebCore/platform/graphics/InbandGenericCue.h:
(WebCore::GenericCueData::GenericCueData):
(WebCore::InbandGenericCue::uniqueId const):
* Source/WebCore/platform/graphics/InbandGenericCueIdentifier.h:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRecordIdentifier.h:
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebURLSchemeHandlerIdentifier.h:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::dataDetectorImageOverlayPositionInformation):

Canonical link: <a href="https://commits.webkit.org/282414@main">https://commits.webkit.org/282414@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/efbb87292397c057cddf7e4c82f84f4fbc25d1be

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63118 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42474 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15714 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67139 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/13726 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/65238 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50161 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14006 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/50861 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9472 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66187 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39445 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54650 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31547 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36130 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/11997 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12598 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/56228 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/57670 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12326 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/68834 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/62361 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7064 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/11937 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58173 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7096 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54722 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58386 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/5890 "Passed tests") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/84124 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9516 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38294 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/14820 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/39374 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/40485 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/39116 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pvollan commented on Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
```diff
@@ -3067,12 +3067,10 @@ static void dataDetectorImageOverlayPositionInformation(const HTMLElement& overl
         return;
 
-    auto identifier = LegacyNullableObjectIdentifier<ImageOverlayDataDetectionResultIdentifierType>(*identifierValue);
-    if (!identifier.isValid())
```
Is the `isValid()` check not needed when using `ObjectIdentifier`?

cdumez replied:
ObjectIdentifier doesn't have an "invalid" state anymore. That's the difference with LegacyNullableObjectIdentifier. You cannot default-construct a ObjectIdentifier or construct it with 0.

cdumez replied:
Note that isValid() was checking that the internal integer wasn't 0. Now I added a `|| !*identifierValue` check above to make sure we can't pass 0 to the ObjectIdentifier constructor.

pvollan reviewed (approved):
R=me.

cdumez committed:
Reduce use of LegacyNullableObjectIdentifier
https://bugs.webkit.org/show_bug.cgi?id=278298

Reviewed by Per Arne Vollan.

Reduce use of LegacyNullableObjectIdentifier, and use ObjectIdentifier instead.

* Source/WTF/wtf/HashMap.h:
(WTF::Y>::takeOptional):
* Source/WebCore/html/track/InbandGenericTextTrack.cpp:
(WebCore::GenericTextTrackCueMap::remove):
* Source/WebCore/page/cocoa/ImageOverlayDataDetectionResultIdentifier.h:
* Source/WebCore/page/mac/ImageOverlayControllerMac.mm:
(WebCore::ImageOverlayController::handleDataDetectorAction):
* Source/WebCore/platform/graphics/ImageDecoderIdentifier.h:
* Source/WebCore/platform/graphics/InbandGenericCue.cpp:
(WebCore::InbandGenericCue::toJSONString const):
* Source/WebCore/platform/graphics/InbandGenericCue.h:
(WebCore::GenericCueData::GenericCueData):
(WebCore::InbandGenericCue::uniqueId const):
* Source/WebCore/platform/graphics/InbandGenericCueIdentifier.h:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRecordIdentifier.h:
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebURLSchemeHandlerIdentifier.h:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::dataDetectorImageOverlayPositionInformation):

Canonical link: https://commits.webkit.org/282414@main
